### ISSUE TITLE
[BUG] Azure search - ObjectDisposedException thrown instead of IndexBatchException when merging index batch 

### ISSUE THREAD

**Issue by cwi3310** [Azure/azure-sdk-for-net#6910]
**Describe the bug**
Using Net Framework 4.7.1, when merging documents with key not present in azure search index, a ObjectDisposedException is thrown instead of IndexBatchException.

***Exception or Stack Trace***
Cannot access a disposed object. Object name: 'System.Net.Http.StringContent'
  at System.Net.Http.HttpContent.CheckDisposed()
   at System.Net.Http.HttpContent.ReadAsStringAsync()
   at Microsoft.Azure.Search.DocumentsOperations.<DoIndexAsync>d__19`1.MoveNext()

**To Reproduce**
1. Create a batch of document complying to an azure search index model but with the key missing in index
2. Merge the batch to the index and catch the exception thrown
The exception thrown should be ObjectDisposedException instead of IndexBatchException

***Code Snippet***
```
var searchClient = new SearchServiceClient(searchName, new SearchCredentials(searchKey));
var documents = new Model[]
{
    new Model
    {
         Id = "abcd",
         Description = "Key Id not present in index"
    },
};
var batch = IndexBatch.Merge(documents);

try
{
    searchClient.Indexes.GetClient(indexName).Documents.Index(batch);
    Console.WriteLine("\t> Batch successfully run");
}
catch (IndexBatchException e)
{
    Console.WriteLine("\t> Batch failed with expected exception");
}
catch (ObjectDisposedException ex)
{
    Console.WriteLine("\t> Batch failed with unexpected exception");
}
```

**Expected behavior**
We expect the IndexBatchException to be thrown instead of ObjectDisposedException

**Setup:**
 - OS: Windows
 - IDE : Visual Studio 2019
 - Microsoft.Azure.Search.Data 9.0.1
   .Net Framework 4.7.1

**Additional context**
The problem doesn't seems to appears in .Net Core.

**Comment by brjohnstmsft** [comment]
@cwi3310 Thanks for reporting this. We'll investigate.

**Comment by brjohnstmsft** [comment]
@arv100kri is investigating

**Comment by arv100kri** [comment]
@cwi3310 - Thanks for reporting the bug, I've checked in a pull request that will fix this.
I'll update this thread, sometime next week once we make a release of a new nuget package.

**Comment by arv100kri** [comment]
We have a new nuget package released (v 9.1.0) that should contain the fixes from the aforementioned PR: https://www.nuget.org/packages/Microsoft.Azure.Search/
